test/controllers/rentals_controller_test.rb

```diff
@@ -0,0 +1,128 @@
+require "test_helper"
+
+describe RentalsController do
+  
+  describe "check out" do
```

Comment:
None of your tests here check for a decrease in available_inventory, which would've caught the problem that the smoke tests surface.

Reply:
...at least that's my assumption about where the bug is based on the error message I'm seeing on the checkin API